include/fiction/algorithms/simulation/sidb/critical_temperature.hpp

```diff
@@ -49,12 +50,19 @@ struct critical_temperature_params
     enum class simulation_engine : uint8_t
```

Comment:
Thanks for adding it to the temperature sim! I would suggest the following: We can get rid of the simulation engine here and just work with ``sidb_simulation_engine.hpp``. Maybe we can even add a ``enum class heuristic_sidb_simulation_engine`` there. What do you think?

Reply:
That sounds better indeed. Would you say implement EXGS too then? No real sense in it though, so better to let it land in `assert(false && "Unsupported simulation engine")`

Reply:
Implemented